src/test/java/seedu/address/logic/parser/AddCommandParserTest.java

```diff
@@ -107,87 +115,116 @@ public void parse_compulsoryFieldMissing_failure() {
         assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
-                        + INVALID_GITHUB_LONGER_THAN_39_DESC + TELEGRAM_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
-                GithubUsername.MESSAGE_CONSTRAINTS);
+                + INVALID_GITHUB_LONGER_THAN_39_DESC + STUDENTID_DESC_BOB + TELEGRAM_DESC_BOB + TAG_DESC_HUSBAND
+                + TAG_DESC_FRIEND, GithubUsername.MESSAGE_CONSTRAINTS);
 
         // invalid telegram
         assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
-                + GITHUB_DESC_BOB + INVALID_TELEGRAM_DESC + TAG_DESC_HUSBAND, Telegram.MESSAGE_CONSTRAINTS);
+                + GITHUB_DESC_BOB + INVALID_TELEGRAM_DESC + STUDENTID_DESC_BOB + TAG_DESC_HUSBAND,
+                Telegram.MESSAGE_CONSTRAINTS);
+
+        // invalid student id (does not start with A)
+        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + INVALID_STUDENTID_START_LETTER_DESC + TAG_DESC_HUSBAND
+                + TAG_DESC_FRIEND, StudentId.MESSAGE_CONSTRAINTS);
+
+        // invalid student id (wrong length)
+        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + INVALID_STUDENTID_WRONG_LENGTH_DESC + TAG_DESC_HUSBAND
+                + TAG_DESC_FRIEND, StudentId.MESSAGE_CONSTRAINTS);
+
+        // invalid student id (does not end with letter)
+        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + INVALID_STUDENTID_END_NUMBER_DESC + TAG_DESC_HUSBAND
+                + TAG_DESC_FRIEND, StudentId.MESSAGE_CONSTRAINTS);
+
+        // invalid student id (has a letter in the middle)
+        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + INVALID_STUDENTID_MIDDLE_LETTER + TAG_DESC_HUSBAND
+                + TAG_DESC_FRIEND, StudentId.MESSAGE_CONSTRAINTS);
 
         // two invalid values, only first invalid value reported
         assertParseFailure(parser, INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB + INVALID_ADDRESS_DESC
-                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB, Name.MESSAGE_CONSTRAINTS);
+                + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + STUDENTID_DESC_BOB, Name.MESSAGE_CONSTRAINTS);
 
         // non-empty preamble
         assertParseFailure(parser, PREAMBLE_NON_EMPTY + NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + GITHUB_DESC_BOB + TELEGRAM_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
+                + ADDRESS_DESC_BOB + GITHUB_DESC_BOB + STUDENTID_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
```

Comment:
Accidentally overwrote the TELEGRAM_DESC_BOB string.

Reply:
Nice catch!